### PR TITLE
fix: check for enterprise in the create/edit templates UI

### PR DESCRIPTION
**Pull request by daveleek**
Hiding with enterprise tag in routes wasn't enough anymore as we're showing those to Pro with upgrade messages. This temporarily disables the UI completely.

Before
Skjermbilde 2024-12-16 kl  11 26 36

After
image

Also disables the create screen if you know the direct url:
Skjermbilde 2024-12-16 kl  14 49 48

### CODE REVIEW

**Review by gastonfournier — COMMENTED** [review]
Can you add screenshots before & after just to understand the PR better?

**Comment by daveleek** [comment]
@gastonfournier thanks, yeah updated description now

**Review by gastonfournier — CHANGES_REQUESTED** [review]
I don't think this improves the UI. If we don't want to show it, we should remove the item in the left side, making the UI blank doesn't seem like a good solution

File: frontend/src/component/releases/ReleaseManagement/ReleaseManagement.tsx [hunk 1]
```diff
@@ -32,7 +40,7 @@ export const ReleaseManagement = () => {
                                 }}
                                 maxWidth='700px'
                                 permission={CREATE_RELEASE_TEMPLATE}
-                                disabled={false}
+                                disabled={!isEnterprise()}
```
Comment (gastonfournier) [inline frontend/src/component/releases/ReleaseManagement/ReleaseManagement.tsx:43]: :+1:

File: frontend/src/component/releases/ReleaseManagement/ReleaseManagement.tsx [hunk 2]
```diff
@@ -10,12 +10,20 @@ import { useNavigate } from 'react-router-dom';
...
 export const ReleaseManagement = () => {
     usePageTitle('Release management');
     const navigate = useNavigate();
     const data = useReleasePlanTemplates();
 
+    const { isEnterprise } = useUiConfig();
+    const releasePlansEnabled = useUiFlag('releasePlans');
+    if (!releasePlansEnabled) {
+        return null;
+    }
```
Comment (gastonfournier) [inline frontend/src/component/releases/ReleaseManagement/ReleaseManagement.tsx:25]: I think this is fine if the feature is disabled you should not be able to see this... I don't think it's needed due to https://github.com/Unleash/unleash/blob/3696420c51dc9d673c6da1393c4d0a61b8c3af24/frontend/src/component/menu/routes.ts#L290-L292 but :shrug:

**Review by gastonfournier — APPROVED** [review]
LG! Thanks for addressing the comment!

**Pull request merged**